subscriber.rb

```diff
@@ -0,0 +1,52 @@
+require 'aws-sdk-sqs'
+require_relative 'load_aws'
+
+class Subscriber
+
+  attr_reader :sqs
+
+  def initialize
+    creds = Aws::Credentials.new(AWS["access_key_id"], AWS["secret_access_key"])
+    @sqs = Aws::SQS::Client.new(region: AWS["region"], credentials: creds)
+  end
+
+# Todo: URL validation
+  def read_queue(options=[])
+    queue_url = options[0].chomp
+    # Note: AWS SQS wait_time_seconds ranges between 0-20 seconds. 
+    wait_time_seconds = if options[1].nil?
+                          0 
+                        elsif options[1].to_i > 20
+                          puts 'Your wait time must be between 0 to 20 seconds.'
+                          exit(1)
+                        else
+                          options[1].to_i
+                        end
+    
+    loop do
+      result = sqs.receive_message(queue_url: queue_url, max_number_of_messages: 1, wait_time_seconds: wait_time_seconds)
+      
+      result.messages.each do |msg|
```

Comment:
Is this loop required if `max_number_of_messages` is `1` ?

Reply:
Even if max number of messages is one, we need the loop since sqs.receive_message always returns an array. So even if we were to access one message, we would need to go through the array.

Reply:
Ok. will do.